components/motor/gpio/motor_encoder.go

```diff
@@ -389,8 +392,9 @@ func (m *EncodedMotor) goForInternal(ctx context.Context, rpm, goalPos, directio
 			}
 
 			// start a new rpmMonitor
+			// create rpmCtx from the motor cancelCtx
 			var rpmCtx context.Context
-			rpmCtx, m.rpmMonitorDone = context.WithCancel(context.Background())
+			rpmCtx, m.rpmMonitorDone = context.WithCancel(m.cancelCtx)
```

Comment:
looking at the current main for EncodedMotor, it looks like the `cancelCtx` was never used before. since this function is effectively meant to be blocking I think we can just use the requests context and remove m.cancelCtx from the state. 

we should be able to use the `opMgr`  on the motor to handle stopping/closing the motor

Reply:
hmm let me try that instead

Reply:
well, I think the context isn't being used to make it blocking, it's just making it so we can cancel the old rpm monitor when we want to start a new one/if we want to interrupt the old one

Reply:
yeah, but we can just cancel the old rpm monitor thru the op manager if we use GoFor's context in rpmMonitor. I haven't seen a reason as to why we need more than one context for a go for call but maybe I'm missing something?

Reply:
oh I see, one sec let me push that change